src/views/Pools/components/PoolCard/CardActions/StakeActions.tsx

```diff
@@ -32,33 +32,41 @@ const StakeAction: React.FC<StakeActionsProps> = ({
   isStaked,
   isLoading = false,
 }) => {
-  const { stakingToken, earningToken, stakingLimit, isFinished } = pool
+  const { stakingToken, stakingLimit, isFinished, userData } = pool
   const { t } = useTranslation()
-  const convertedLimit = getDecimalAmount(new BigNumber(stakingLimit), earningToken.decimals)
-  const stakingMax =
-    stakingLimit && stakingTokenBalance.isGreaterThan(convertedLimit) ? convertedLimit : stakingTokenBalance
   const stakedTokenBalance = getBalanceNumber(stakedBalance, stakingToken.decimals)
   const stakedTokenDollarBalance = getBalanceNumber(
     stakedBalance.multipliedBy(stakingTokenPrice),
     stakingToken.decimals,
   )
-
   const [onPresentTokenRequired] = useModal(<NotEnoughTokensModal tokenSymbol={stakingToken.symbol} />)
 
   const [onPresentStake] = useModal(
-    <StakeModal stakingMax={stakingMax} isBnbPool={isBnbPool} pool={pool} stakingTokenPrice={stakingTokenPrice} />,
+    <StakeModal
+      isBnbPool={isBnbPool}
+      pool={pool}
+      stakingTokenBalance={stakingTokenBalance}
+      stakingTokenPrice={stakingTokenPrice}
+    />,
   )
 
   const [onPresentUnstake] = useModal(
     <StakeModal
-      stakingMax={stakedBalance}
```

Comment:
Ah fuck. This is the crucial bit ahaha. `stakingMax` was different depending on whether it was a staking/unstaking action.